CRM/Admin/Form/Setting.php

```diff
@@ -133,7 +133,12 @@ public function buildQuickForm() {
           $this->add('date', $setting, ts($props['title']), CRM_Core_SelectValues::date(NULL, 'M d'));
         }
         else {
-          $this->$add($setting, ts($props['title']));
+          if (CRM_Core_BAO_Setting::isEnvironmentSet($setting)) {
```

Comment:
This seems like bad form. Why is this one setting being given special one-off treatment when other settings are dealt with in a general way?